std/algorithm/searching.d

```diff
@@ -799,15 +802,33 @@ if (isForwardRange!R
                   haystack.popFront();
             }
         }
+        //Because of @@@8804@@@: Avoids both "unreachable code" or "no return statement"
+        static if (isInfinite!R) assert(0);
+        else return -1;
     }
     else
     {
+        static struct Result
+        {
+            ptrdiff_t steps, needle; // both -1 when nothing was found
+            alias steps this; // compatible with previous ptrdiff_t return type
+            ptrdiff_t opIndex(size_t idx) // faking a tuple
```

Comment:
I'm playing with the idea to change the signature to `ptrdiff_t opIndex(bool idx)`. Due to implicit conversion, `res[0]` and `res[1]` would still work. Looks like a hack, but since this type models `Tuple!(ptrdiff_t, "steps", ptrdiff_t needle)` and tuples do not satisfy `isRandomAccessRange`, we don't need to use `size_t` either. Using `std.typecons.Flag` could also work, but I'm failing to find a good name for the template parameter.

Alternatively, if we had [DIP66](https://wiki.dlang.org/DIP66) implemented, we could do:
```D
struct Result
{
    import std.meta : AliasSeq;
    
    alias Members = AliasSeq!(ptrdiff_t, ptrdiff_t);
    
    Members _members;
    
    alias steps = _members[0];
    alias needle = _members[1];
    
    alias _members this; // 1)
	alias steps this; // 2)
}

unittest
{
    auto r = Result(1, 2);
    assert(r[0] == 1); // 1)
    assert(r[1] == 2); // 1)
    assert(r == 1); // 2)
}
```

But unfortunately you can only choose between 1) or 2). (We also can't use `T opCast(T : ptrdiff_t)() { return steps; }`, because this requires the use of explicit casting - such as `assert(cast(ptrdiff_t)r == 1)` - and so it would be a breaking change.)